doc/api/process.md

```diff
@@ -1881,6 +1881,331 @@ a code.
 Specifying a code to [`process.exit(code)`][`process.exit()`] will override any
 previous setting of `process.exitCode`.
 
+## `process.finalization.register(ref, callback)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1.1 - Active Development
+
+* `ref` {Object | Function} The reference to the resource that is being tracked.
+* `callback` {Function} The callback function to be called when the resource
+  is finalized.
+  * `ref` {Object | Function} The reference to the resource that is being tracked.
+  * `event` {string} The event that triggered the finalization. Defaults to 'exit'.
+
+This function registers a callback to be called when the process emits the `exit` event.
+
+If the object is garbage collected before `exit` event is emitted, the callback will be
+removed from the finalization registry, and it will not be called on process exit.
+
+Inside the callback you can do whatever you want with the object,
+including the release of resources allocated by that resource.
+
+The idea of ​​this function is to help you free up resources when the process exit,
+but also let the object be garbage collected if it is no longer being used.
+
+Ex: you can register a stream to be released when the process exit, if the stream is
+garbage collected before the process exit, we no longer need to release the stream, so in this case
+we just remove the callback from the finalization registry.
```

Comment:
All of them if you don't want to lose data after a `process.exit(1)` is called.
Even `console.log()` might lose data.

(The fact that I had to create a new write stream implementation to solve this problem shows that nobody cared for this shutdown problem).

Reply:
@mcollina What you are describing sounds like _flushing_ a stream, and that's not what I would guess based on the current wording.

On a side note, applications should generally explicitly flush streams when necessary. I'm surprised that Node.js isn't flushing `console.log()` before exiting.

Reply:
English is not my native language so that's why the wording is a little bad sometimes, I really appreciate your comments, as soon I have some time I will update the docs based on your suggestions and change the dispose to flush, which describes better the intention here.

Reply:
> I'm surprised that Node.js isn't flushing console.log() before exiting.

Same when I found out.

Note that our streams don't
have a flush() function, so calling "flushing a stream" might be incorrect as well.